modules/rostests/apitests/wscript/wsf.c

```diff
@@ -0,0 +1,166 @@
+/*
+ * PROJECT:     ReactOS API tests
+ * LICENSE:     LGPL-2.1+ (https://spdx.org/licenses/LGPL-2.1+)
+ * PURPOSE:     Test for wscript.exe
+ * COPYRIGHT:   ReactOS Team
+ */
+
+#include <apitest.h>
+#include <windows.h>
+#include <shlwapi.h>
+
+#define MYGUID "{898AC78E-BFC7-41FF-937D-EDD01E666707}"
+
+static DWORD getregdw(HKEY hKey, LPCSTR sub, LPCSTR name, DWORD *out, DWORD defval)
+{
+    DWORD size = sizeof(*out);
+    *out = 0;
+    LRESULT e = SHGetValueA(hKey, sub, name, NULL, out, &size);
+    if (e)
+        *out = defval;
+    return e;
+}
+
+static BOOL makestringfile(LPWSTR path, LPCSTR ext, LPCSTR string, const BYTE *map)
+{
+    UINT cch = GetTempPathW(MAX_PATH, path);
+    UINT16 i = 0;
+    while (++i)
+    {
+        HANDLE hFile;
+        wsprintfW(path + cch, L"~%u.%hs", i, ext ? ext : "tmp");
```

Comment:
You should be able to use [GetTempFileName](https://learn.microsoft.com/en-us/windows/win32/api/fileapi/nf-fileapi-gettempfilenamew) since you're here creating the file.

Reply:
GetTempFileName always uses the .tmp extension but here we need a specific extension because wscript.exe uses the extension to detect the script type.